code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/intermediate_model_validation/ContainerBaseIntermediateModelValidator.xtend

```diff
@@ -0,0 +1,54 @@
+package de.fhdo.lemma.model_processing.code_generation.container_base.intermediate_model_validation
+
+import de.fhdo.lemma.model_processing.annotations.IntermediateModelValidator
+import de.fhdo.lemma.model_processing.builtin_phases.intermediate_model_validation.AbstractIntermediateDeclarativeValidator
+import de.fhdo.lemma.operation.intermediate.IntermediatePackage
+import de.fhdo.lemma.operation.intermediate.IntermediateContainer
+import de.fhdo.lemma.operation.intermediate.IntermediateInfrastructureNode
+import org.eclipse.xtext.validation.Check
+
+/**
+ * The container base intermediate model validator is responsible for checking general aspects for
+ * the container base code generator.
+ *
+ * @author Philip Wizenty(philip.wizenty@fh-dortmund.de)
+ */
+@IntermediateModelValidator
+class ContainerBaseIntermediateModelValidator extends AbstractIntermediateDeclarativeValidator {
+
+    /*
+     * Get the namespace for the intermediate operation model
+     */
+    override getLanguageNamespace() {
+        return IntermediatePackage.eNS_URI
+    }
+
+    /*
+     * Check if container uses the container_base technology which is supported by the
```

Comment:
Actually this is Kubernetes-specific and not for a "generic" `container_base technology`.

Reply:
The Container_base technology refers to the technology stacks Docker, Docker-Compose and Kubernetes. I had difficulties to find a suitable name for it.